### ISSUE TITLE
Filtering not working for me

### ISSUE THREAD

**Issue by drummy2** [ethanresnick/json-api#179]
I am attempting to do filtering in my application.  When making a request using postman I get this response using the below query.

> /categories?filter=(displayLabel,'Getting Started')

> {
    "errors": [
        {
            "detail": "Invalid filter syntax: Expected field expression but \"%\" found. See jsonapi.js.org for details.",
            "source": {
                "parameter": "filter"
            },
            "status": "400",
            "title": "One or more of the query parameters you provided has an invalid value.",
            "code": "https://jsonapi.js.org/errors/invalid-query-param-value"
        }
    ]
}

Any ideas where I am going wrong because I feel it is the same as the README

**Comment by ethanresnick** [comment]
You need to put "Getting Started" between backticks, not single quotes. I.e.,:

```
/categories?filter=(displayLabel,`Getting Started`)
```

**Comment by drummy2** [comment]
Apologies, I pasted one of my many permutations attempting to get filtering to work.  Even with back ticks it still give me the same error. Any help is appreciated.

![postman_test_12](https://user-images.githubusercontent.com/35729186/47073419-2c46e500-d1f0-11e8-9805-c1566e9ccb4b.png)

**Comment by ethanresnick** [comment]
This may be an issue with your HTTP client. Can you try opening the same url in a web browser? What app are you using to test?

**Comment by ethanresnick** [comment]
From the README:

> Be aware that some clients, by default, automatically percent-encode certain characters in url query strings -- especially the backtick and square brackets. This will cause the server to error, because it won't see the appropriate (unescaped) delimiter characters. Whether the client is correct to do this automatic encoding is a nuanced question, as there are competing and conflicting standards governing URLs today (namely, RFC 3986 and WHATWG Url). If you encounter this problem, check your client for a way to make requests without any automatic encoding. If your client absolutely insists on URL encoding backticks (the trickiest character), you can delimit your strings with exclamation points (!) instead, and then encode exclamation points within the string.

**Comment by drummy2** [comment]
I did look and I think it says postman doesn't automatically encode/decode URLS.  I will need to change our ember app and test with that.  Seems to work with Firefox.

**Comment by ethanresnick** [comment]
> I did look and I think it says postman doesn't automatically encode/decode URLS.

Postman may be lying :P If you're getting this error there but not in FF, it's because of encoding. I'm gonna close this, since it's a known concern, and the `!` will work for now if you need to test with Postman etc. (Note: you can use the backticks in Ember and the exclamation points for Postman if you really want; they're identical.)

**Comment by PareshSolankiMI** [comment]
@ethanresnick also I got issue in Firefox

**Comment by ethanresnick** [comment]
@PareshSolankiMI What is the filter string you're trying? Have you tried using the `!` delimited strings instead of backtick ones?

**Comment by PareshSolankiMI** [comment]
> @PareshSolankiMI What is the filter string you're trying? Have you tried using the `!` delimited strings instead of backtick ones?

Yes I tried with it. But I got error like: Invalid filter syntax: input.charCodeAt is not a function See jsonapi.js.org for details.

So can you please help me on it?
